Mathlib/AlgebraicTopology/SimplicialSet/Path.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2024 Emily Riehl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Emily Riehl, Joël Riou
+-/
+import Mathlib.AlgebraicTopology.SimplicialSet.Basic
+
+/-!
+# Paths in simplicial sets
+
+A path in a simplicial set `X` of length `n` is a directed path comprised of `n + 1` 0-simplices
+and `n` 1-simplices, together with identifications between 0-simplices and the sources and targets
+of the 1-simplices.
+
+An `n`-simplex has a maximal path, the `spine` of the simplex, which is a path of length `n`.
+-/
+
+universe v u
+
+namespace SSet
+
+open CategoryTheory
+
+open Simplicial SimplexCategory
+
+variable (X : SSet.{u})
+
+/-- A path in a simplicial set `X` of length `n` is a directed path of `n` edges.-/
+@[ext]
+structure Path (n : ℕ) where
+  /-- A path includes the data of `n+1` 0-simplices in `X`.-/
+  vertex (i : Fin (n + 1)) : X _[0]
+  /-- A path includes the data of `n` 1-simplices in `X`.-/
+  arrow (i : Fin n) : X _[1]
+  /-- The sources of the 1-simplices in a path are identified with appropriate 0-simplices.-/
+  arrow_src (i : Fin n) : X.δ 1 (arrow i) = vertex i.castSucc
+  /-- The targets of the 1-simplices in a path are identified with appropriate 0-simplices.-/
+  arrow_tgt (i : Fin n) : X.δ 0 (arrow i) = vertex i.succ
+
+
+variable {X} in
+/-- For `j ≤ k ≤ n`, a path of length `n` restricts to a path of length `k-j`, namely the subpath
+spanned by the vertices `j ≤ i ≤ k` and edges `j ≤ i < k`. -/
+def Path.interval {n : ℕ} (f : Path X n) (j l : ℕ) (hjl : j + l < n + 1) :
```

Comment:
Is there a particular reason you choose for `< n + 1`? Intuitively, I would have expected `≤ n`
```suggestion
def Path.interval {n : ℕ} (f : Path X n) (j l : ℕ) (hjl : j + l ≤ n) :
```

Reply:
@jcommelin I had similar instincts but I choose less than rather than less than or equal to to match the field used in `Fin n` namely `isLt : LT.lt val n`.

Reply:
Made this change.